cylc/flow/data_store_mgr.py

```diff
@@ -188,6 +192,30 @@ def task_mean_elapsed_time(tdef):
     return tdef.rtconfig.get('execution time limit', None)
 
 
+def runtime_from_config(rtconfig):
+    """Populate runtime object from config."""
+    try:
+        platform = rtconfig['platform']['name']
+    except (KeyError, TypeError):
```

Comment:
Shouldn't `rtconfig['platform']` just be a string of the platform name? In what circumstance would it be a dict containing a `name` key?

Reply:
The confusing thing is.. Platform gets changed before job submission, both exist.

The dict version appears to be more common..

Reply:
That is confusing. Is it just because one of your calls to `runtime_from_config` passes in the job config instead of the `taskdef.rtconfig`?

Reply:
Ping @dwsutherland - one question to respond to here. As I recall, I didn't think it should be like this, hence my suggested reason for it.

Reply:
Yes, the platform gets fully resolved/loaded on job construction.. however it's just a string in the workflow config runtime.